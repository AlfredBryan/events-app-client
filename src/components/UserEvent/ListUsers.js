import React, { Component } from "react";
import { Link } from "react-router-dom";
import axios from "axios";

import "./UserE.css";
import NavBar from "../NavBar/NavBar";

const apiURL = "http://localhost:4000/api/events";

class ListUsers extends Component {
  constructor(props) {
    super(props);
    this.state = {
      eventId: this.props.match.params,
      users: []
    };
  }

  componentDidMount() {
    let eventId = this.state.eventId;
    let id = eventId.id;
    axios.get(`${apiURL}/${id}`).then(res => {
      this.setState({
        users: res.data.Users
      });
    });
  }

  render() {
    let { users } = this.state;
    let { eventId } = this.state;
    let id = eventId.id;
    if (users) {
      return (
        <React.Fragment>
          <NavBar />
          <div className="user-main links">
            <div className="container">
              <h5>Users Signed Up to Event</h5>
              <Link to={`/event/signup/${id}`}>
                <p>click to signup for event</p>
              </Link>
              {users.map(user => (
                <div key={user.id} className="d-flex wrap">
                  <p className="fname">{user.firstName}</p>
                  <p className="lname">{user.lastName}</p>
                </div>
              ))}
            </div>
          </div>
        </React.Fragment>
      );
    } else {
      return (
        <React.Fragment>
          <NavBar />
          <div className="container none-yet">
            <div className="row">
              <div className="col-10 mx-auto text-center text-title pt-5">
                <h2>No SignUp to Event Yet</h2>
                <Link to={`/event/signup/${id}`}>
                  <p>click to signup for event</p>
                </Link>
              </div>
            </div>
          </div>
        </React.Fragment>
      );
    }
  }
}

export default ListUsers;
