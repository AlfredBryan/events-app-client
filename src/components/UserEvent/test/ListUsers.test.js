import React from "react";
import { shallow } from "enzyme";
import ListUsers from "../ListUsers";

jest.mock("axios", () => {
  const testUsers = {
    users: []
  };

  return {
    get: jest.fn(() => Promise.resolve(testUsers))
  };
});

const match = {
  params: {
    eventId: 678256526 //any id you want to set
  }
};

const axios = require("axios");

it("Fetch registered Users #componentDidMount", done => {
  const wrapper = shallow(<ListUsers match={match} />);
  wrapper.instance().componentDidMount();
  expect(axios.get).toHaveBeenCalled();
  expect(axios.get).toHaveBeenCalledWith(
    "https://events-apps.herokuapp.com/api/events/undefined"
  );
  expect(wrapper.state()).toMatchSnapshot("users", []);
  done();
});
