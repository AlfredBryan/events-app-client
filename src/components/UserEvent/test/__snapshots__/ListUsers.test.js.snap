// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fetch registered Users #componentDidMount: users 1`] = `
Object {
  "eventId": Object {
    "eventId": 678256526,
  },
  "users": Array [],
}
`;
